crmd/remote_lrmd_ra.c

```diff
@@ -1173,4 +1189,43 @@ remote_ra_process_pseudo(xmlNode *xml)
         }
     }
     freeXpathObject(search);
+
+    /* if we get a list of nodes in maintenance with the transition
+     * we clear maintenance in all remote-nodes and just
+     * set it for the nodes given
+     */
+    search = xpath_search(xml, XPATH_PSEUDO_MAINTENANCE_NODES);
+    crm_trace("Transition holds %d nodes in maintenance-mode",
+              numXpathResults(search));
+    if (numXpathResults(search)) {
+        lrm_state_t *lrm_state;
+        int result;
+
+        lrm_state_foreach_remote(reset_maintenance, NULL);
+
+        for (result = 0; result < numXpathResults(search); result++) {
+            lrm_state = lrm_state_find(ID(getXpathResult(search, result)));
+
+            if (lrm_state && lrm_state->remote_ra_data) {
+                remote_ra_maintenance(lrm_state, TRUE);
+            }
+        }
+    }
+    freeXpathObject(search);
 }
+
+void
+remote_ra_maintenance(lrm_state_t * lrm_state, gboolean maintenance)
+{
+    remote_ra_data_t *ra_data = lrm_state->remote_ra_data;
+
+    ra_data->is_maintenance = maintenance;
+}
+
+gboolean
+remote_ra_is_in_maintenance(lrm_state_t * lrm_state)
+{
+    remote_ra_data_t *ra_data = lrm_state->remote_ra_data;
+
+    return ra_data->is_maintenance;
+}
```

Comment:
Ah, good point about remote_ra_data_t. Probably a good idea to keep it where it is.

When we do inline functions, we always put them static inline in their entirety in the header. That's the only way that's portable across all common compilers. Good compilers with optimization enabled will do a better job of deciding when to inline than a human coder will, so it's mainly to address less capable compilers and when someone turns optimization off (e.g. for debug symbols). A minor concern.